### DISCUSSION TITLE
WiX v4 MsiProperty Issues

### DISCUSSION THREAD

**Discussion by DarrelF** [wixtoolset/issues#7791] [Questions]
The MsiProperty documentation for WiX v4 ([https://wixtoolset.org/docs/schema/wxs/msiproperty/]) documents the following:

`Name (String, required) : The name of the MSI property to set. Burn controls the follow MSI properties so they cannot be set with MsiProperty: ACTION, ALLUSERS, REBOOT, REINSTALL, REINSTALLMODE`
`
This suggests property REMOVE is legal; however, I get the following error with WiX v4 when using REMOVE:

`Severity	Code	Description	Project	File	Line	Suppression State
Error	WIX0365	The 'REMOVE' MsiProperty is controlled by the bootstrapper and cannot be authored. (Illegal properties are: 'ACTION', 'ADDLOCAL', 'ADDSOURCE', 'ADDDEFAULT', 'ADVERTISE', 'ALLUSERS', 'REBOOT', 'REINSTALL', 'REINSTALLMODE', or 'REMOVE'.) Remove the MsiProperty element.
`

In WiX v3, REMOVE was a valid Name for MsiProperty and the open source MSI for MariaDB documents using it.
Has REMOVE and the other "now illegal properties" been made illegal by design?

Also, MsiProperty does not allow an empty value (i.e., Value=""):

`Severity	Code	Description	Project	File	Line	Suppression State
Error	WIX0006	The MsiProperty/@Value attribute's value cannot be an empty string. If a value is not required, simply remove the entire attribute.
`

The MariaDB MSI has a property named CLEANUPDATA that must be specified as an empty value to have the uninstall not remove the data/database and MsiProperty won't allow Value="". 

Is it technically not possible to allow MsiProperty/@Value to be an empty string?

**Comment by barnson** [comment]
1. Yes, it's by design: https://wixtoolset.org/docs/development/wips/ba-configurable-reinstallmode/
2. It's probably possible. You can file a feature request for the triage team to evaluate.
